src/natcap/invest/utils.py

```diff
@@ -43,6 +44,205 @@
 # leaves Projected CRS alone
 DEFAULT_OSR_AXIS_MAPPING_STRATEGY = osr.OAMS_TRADITIONAL_GIS_ORDER
 
+# the same unit registry instance should be shared across everything
+u = pint.UnitRegistry()
+# see https://pint.readthedocs.io/en/stable/defining.html
+# pint doesn't allow multiple base units for the same dimension
+# but you can define mutliple dimensionless units with []
+# https://github.com/hgrecco/pint/issues/1278
+u.define('currency = [value]')    # non-specific unit of value
+u.define('pixel = []')            # non-specific unit of area
+u.define('projection_unit = []')  # non-specific unit of length
+# used in coastal vulnerability, the DEM pixel values measure
+# elevation (length) but the specific units don't matter
+u.define('linear_unit_of_elevation = []')  # non-specific unit of length
+# add "us_survey_foot" on to the aliases because it's used in some rasters
+u.define('survey_foot = 1200 / 3937 * meter = sft = us_survey_foot')
+# Vitamin A in the crop production nutrient table is measured in IUs
+# A special unit in pharmacology that measures biologically active substances
+# May be converted to weight or volume, but conversion factors are specific
+# to the substance. I couldn't find a definition of its dimensionality.
+u.define('international_unit = [biologic_amount] = iu = IU')
+# Use u.none for unitless measurements
+u.define('none = []')
+
+WORKSPACE_SPEC = {
+    "name": "Workspace",
+    "about": (
+        "The folder where all intermediate and output files of the model "
+        "will be written.  If this folder does not exist, it will be "
+        "created."),
+    "type": "directory",
+    "contents": {},
+    "validation_options": {
+        "exists": False,
+        "permissions": "rwx",
```

Comment:
Oh interesting. I'm curious to see how `permissions` is used. I guess this is also assuming `rwx` at the user level.